File: backend-java/src/main/java/com/skewwhiffy/auraltester/dao/NoteDao.java
package com.skewwhiffy.auraltester.dao;

import com.skewwhiffy.auraltester.notation.model.note.Note;

public record NoteDao(String noteName, AccidentalDao accidental) {
    public Note toModel() {
        return new Note(noteName, accidental.toModel());
    }
}

File: backend.old/src/main/kotlin/com/skewwhiffy/auraltester/dao/NoteDao.kt
package com.skewwhiffy.auraltester.dao

import com.skewwhiffy.auraltester.notation.model.note.Note

data class NoteDao(
    val noteName: String,
    val accidental: AccidentalDao
) {
    val model
        get() = Note(noteName, accidental.model)
}
